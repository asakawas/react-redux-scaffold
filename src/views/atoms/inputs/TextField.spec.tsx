import React from 'react';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import TextField from './TextField';

describe('TextField', () => {
  test('is rendered as expected', () => {
    const { asFragment } = render(<TextField name="test" label="Test" />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('readOnly ver is rendered as expected', () => {
    const { asFragment } = render(<TextField name="test" label="Test" readOnly />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('can be typed in if readOnly is false', async () => {
    const renderResult = render(<TextField name="test" label="Test" />);

    const tree = renderResult.getByLabelText('input-test');
    // @ts-ignore  Property 'value' does not exist on type 'HTMLElement'
    expect(tree.value).toBe('');
    await userEvent.type(tree, 'test test');
    // @ts-ignore  Property 'value' does not exist on type 'HTMLElement'
    expect(tree.value).toBe('test test');
  });

  test('can not be typed in if readOnly is true', async () => {
    const renderResult = render(<TextField name="test" label="Test" readOnly />);

    const tree = renderResult.getByLabelText('input-test');
    // @ts-ignore  Property 'value' does not exist on type 'HTMLElement'
    expect(tree.value).toBe('');
    await userEvent.type(tree, 'test test');
    // @ts-ignore  Property 'value' does not exist on type 'HTMLElement'
    expect(tree.value).toBe('');
  });

  test("doesn't has a helper text to show 'Read-only' if readOnly is false", () => {
    const renderResult = render(<TextField name="test" label="Test" />);

    const tree = renderResult.queryByText('Read-only');
    expect(tree).toBeNull();
  });

  test('has a helper text to show "Read-only" if readOnly is true', () => {
    const renderResult = render(<TextField name="test" label="Test" readOnly />);

    const tree = renderResult.queryByText('Read-only');
    expect(tree).not.toBeNull();
  });
});
