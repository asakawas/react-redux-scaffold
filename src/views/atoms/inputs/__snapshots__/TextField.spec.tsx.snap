// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField is rendered as expected 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiTextField-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
      data-shrink="false"
      for="test"
      id="test-label"
    >
      Test
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    >
      <input
        aria-invalid="false"
        aria-label="input-test"
        class="MuiInputBase-input MuiInput-input"
        id="test"
        name="test"
        type="text"
        value=""
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`TextField readOnly ver is rendered as expected 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiTextField-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
      data-shrink="false"
      for="test"
      id="test-label"
    >
      Test
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    >
      <input
        aria-describedby="test-helper-text"
        aria-invalid="false"
        aria-label="input-test"
        class="MuiInputBase-input MuiInput-input"
        id="test"
        name="test"
        readonly=""
        type="text"
        value=""
      />
    </div>
    <p
      class="MuiFormHelperText-root"
      id="test-helper-text"
    >
      Read-only
    </p>
  </div>
</DocumentFragment>
`;
